src/peergos/shared/user/fs/FileWrapper.java

```diff
@@ -417,31 +395,34 @@ public SymmetricKey getParentKey() {
                                 .collect(Collectors.toSet()))));
     }
 
-    private static CompletableFuture<Set<FileWrapper>> getFilesWithSameWriter(PublicKeyHash owner,
-                                                                              Set<AbsoluteCapability> caps,
-                                                                              Optional<SigningPrivateKeyAndPublicHash> entryWriter,
-                                                                              String ownername,
-                                                                              NetworkAccess network,
-                                                                              Snapshot version) {
-        Set<PublicKeyHash> childWriters = caps.stream()
-                .map(c -> c.writer)
-                .collect(Collectors.toSet());
-        return version.withWriters(owner, childWriters, network)
-                .thenCompose(fullVersion -> network.retrieveAllMetadata(new ArrayList<>(caps), fullVersion)
-                        .thenCompose(rcs -> Futures.combineAll(rcs.stream()
-                                .filter(rc -> ! rc.getProperties().isLink)
-                                .map(rc -> Futures.of(new FileWrapper(rc, Optional.empty(), entryWriter, ownername, fullVersion)))
-                                .collect(Collectors.toSet()))));
-    }
-
     @JsMethod
     public CompletableFuture<Optional<FileWrapper>> getChild(String name, Hasher hasher, NetworkAccess network) {
         return getChild(version, name, hasher, network);
     }
 
-    public CompletableFuture<Optional<FileWrapper>> getChild(Snapshot version, String name, Hasher hasher, NetworkAccess network) {
-        return getChildren(version, hasher, network)
-                .thenApply(children -> children.stream().filter(f -> f.getName().equals(name)).findAny());
+    public CompletableFuture<Optional<FileWrapper>> getChild(Snapshot version,
```

Comment:
I think we also need a getLatestChild() method in case of a concurrent modification by another

Reply:
I'm not sure what you mean. You can always call getUpdated on the child if desired?